Source/WebCore/svg/SVGFilterElement.cpp

```diff
@@ -81,18 +82,34 @@ void SVGFilterElement::attributeChanged(const QualifiedName& name, const AtomStr
             Ref { m_primitiveUnits }->setBaseValInternal<SVGUnitTypes::SVGUnitType>(propertyValue);
         break;
     }
-    case AttributeNames::xAttr:
-        Ref { m_x }->setBaseValInternal(SVGLengthValue::construct(SVGLengthMode::Width, newValue, parseError));
+    case AttributeNames::xAttr: {
+        auto length = SVGLengthValue::construct(SVGLengthMode::Width, newValue, parseError);
+        if (parseError != NoError || newValue.isNull())
+            length = SVGLengthValue(SVGLengthMode::Width, "-10%"_s);
+        Ref { m_x }->setBaseValInternal(length);
```

Comment:
This code is repeated multiple times in this PR. It parses an `SVGLengthValue` if it is invalid it sets the baseVal of the `SVGAnimatedProperty` to the initial value. Should not this code be put in one function so this change is smaller?

Reply:
It is something currently done as well, so I opted for this. I am OK with using helper function or reduce repetition but I was thinking of doing it as future follow-up.

Reply:
I think using table / data driven as Sam suggested or a single function to set the properties should be a better approach. Adding repeated code in multiple places and then trying to shorten it later does not seem an efficient way to do things.

Reply:
I agree that we should include the helper function in the initial patch. The pattern is simple enough that I expect it can be done cleanly.

Reply:
I agree with @darinadler and @karlcow ! This will be much easier to read and reason about.